hoki/spec.py

```diff
@@ -18,3 +22,141 @@ def dopcor(df, z, wl_col_index=0):
+            return custom
+        except FileNotFoundError:
+            raise
+
+def apply_bandpass(spectrum, bandpass):
+    """Applies a bandpass filter to a given spectrum.
+
+    This function overrides the optmized wavelength binning of pysynphot.
+    Instead it uses the wavelength binning of the given spectrum.
+
+    Parameters
+    ----------
+    spectrum : pysynphot Source Spectrum
+        The spectrum of a source
+    bandpass : pysynphot ObsBandpass
+        A bandpass to apply to the spectrum
+
+    Returns
+    -------
+    pysynphot Observation
+        The spectrum with the bandpass applied.
+    """
+
+    return psp.Observation(spectrum, bandpass, binset=spectrum.wave)
+
+
+def get_spectrum(wavelength, flux):
+    """Creats a pysynphot ArraySpectrum object from a BPASS spectrum.
+
+    Parameters
+    ----------
+    wavelength : array/Pandas Series
+        An array of wavelength from BPASS
+    flux : array/Pandas Series
+        An array containing the flux at each wavelength in units
+        Solar luminosity per Angstrom.
+
+    Returns
+    -------
+    pysynphot ArraySpectrum
+        An pysynphot ArraySpectrum with the flux in erg/s/Anstrom.
+    """
+    # check if values are pandas series
+    if type(wavelength) == pd.Series:
+        wavelength = wavelength.values
+    if type(flux) == pd.Series:
+        flux = flux.values
+    return psp.ArraySpectrum(wavelength, flux*3.846e33, fluxunits="Flam")
+
+
+def flux_to_vegamag(bandpass, observation, distance=10):
+    """Calculate a vega magnitude using the bandpass, flux, and pysynphot
+    build in vega spectrum.
+
+    Parameters
+    ----------
+    bandpass : pysynphot BandPass
+        A pysynphot BandPass used to calculate the flux.
+    flux : float
+        The flux in units ergs/s
+    distance : float
+        The distance to the object in pc. DEFAULT=10
+
+    Returns
+    -------
+    float
+        Vega Magnitude
+    """
+    v = psp.Observation(psp.Vega, bandpass, binset=observation.binwave)
+    if (np.diff(observation.binwave) == 1).all(): # check if sum is allowed
+        zp = sum(v.binflux)
+        luminosity = sum(observation.binflux)
+    else:
+        print(f"{observation} does not have width 1 bins."
+              f"Using numpy trapz to calculate the magnitude."
+        )
+        zp = np.trapz(v.binflux, v.binwave)
+        luminosity = np.trapz(observation.binflux, observation.binwave)
+    d = distance*3.0857e18
+    flux = luminosity/(4*np.pi*d**2)
+    return -2.5 * np.log10(flux/zp)
+
+
+def import_custom_filter(filename):
+    """Gives a filter profile class from a filter profile file containing
+    a VOTable.
+
+    Parameters
+    ----------
+    filename : str
+        The filename conatining a VOTable with the filter profile.
+
+    Returns
+    -------
+    pysynphot Bandpass
+        A bandpass filter object from pySynPhot
+    """
+    votable = astropy.io.votable.parse(filename).get_first_table()
```

Comment:
does this fail elegantly  and informatively when a bad filename is given?

Reply:
This fails with a FileNotFoundError if a file is not found. This is as expected.